components/director/internal/formationmapping/handler.go

```diff
@@ -300,8 +362,20 @@ func (b RequestBody) Validate() error {
 	return nil
 }
 
+// Validate validates the formation's request body input
+func (b FormationRequestBody) Validate() error {
+	var fieldRules []*validation.FieldRules
+	fieldRules = append(fieldRules, validation.Field(&b.State, validation.In(model.ReadyFormationState, model.CreateErrorFormationState, model.DeleteErrorFormationState)))
+
+	if b.Error != "" {
+		fieldRules = append(fieldRules, validation.Field(&b.State, validation.In(model.CreateErrorFormationState, model.DeleteErrorFormationState)))
```

Comment:
You are adding those states to the field rules when an error is present, this means that if error is present you allow all 5 states.

I believe the correct way to implement this is:
```go
func (b FormationRequestBody) Validate() error {
	return validation.ValidateStruct(&b,
		validation.Field(&b.State,
			validation.When(len(b.Error) > 0, validation.In(model.CreateErrorFormationState, model.DeleteErrorFormationState)),
			validation.When(len(b.Error) == 0, validation.In(model.ReadyFormationState, model.CreateErrorFormationState, model.DeleteErrorFormationState))))
}
```

Maybe this is valid for the FA validation as well.